source/arborx/access_traits.cc

```diff
@@ -21,22 +21,31 @@ DEAL_II_NAMESPACE_OPEN
 
 namespace ArborXWrappers
 {
+  namespace
+  {
+    template <int dim, typename Number>
+    dealii::Point<3, float>
+    to_3d_float_point(dealii::Point<dim, Number> p)
+    {
+      static_assert(dim >= 1 && dim <= 3);
+      if (dim == 1)
+        return {float(p[0]), 0.f, 0.f};
+      else if (dim == 2)
+        return {float(p[0]), float(p[1]), 0.f};
+      else
+        return {float(p[0]), float(p[1]), float(p[2])};
```

Comment:
I don't think you actually need all of these explicit conversions. `Point<3,Number>` has a constructor that takes three `Number` objects -- which means the conversion can be implicit:
```suggestion
      if (dim == 1)
        return Point<3,float>(p[0], 0.f, 0.f);
      else if (dim == 2)
        return Point<3,float>(p[0], p[1], 0.f);
      else
        return Point<3,float>(p[0], p[1], p[2]);
```